relays/client-substrate/src/finality_source.rs

```diff
@@ -31,22 +32,27 @@ use std::{marker::PhantomData, pin::Pin};
 
 /// Wrapped raw Justification.
 #[derive(Debug, Clone)]
-pub struct Justification<Number> {
+pub struct Justification<H: HeaderT> {
```

Comment:
Is this struct still necessary? The doc comment says that `target_header_number` is decoded from the `raw_justification`, seems that you can get what you need from `GrandpaJustification.commit.target_number`. Please remove.

Reply:
It is required for `impl<H: HeaderT> FinalityProof<H::Number> for Justification<H>` (`FinalityProof` is from the crate that knows nothing about GRANDPA and `GrandpaJustification` is from the crate that shall not reference the `finality-relay` crate). Alternatively, we may extract `FinalityProof` trait to the `bp-header-chain` and impl it for the `GrandpaJustification` there. But eventually I hope that justification will be exposed from `sp-finality-grandpa` && we will be able to remove all our justification-related copypaste => so we'll need this wrapper again.

Reply:
I ended up removing the wrapper despite your last point Slava. I'm not sure when, or even if, we'll get the primitives exposed via `sp-finality-grandpa`. 

We can revisit this if that happens.